model/riscv_mem.sail

```diff
@@ -66,28 +79,39 @@ function phys_mem_read forall 'n, 0 < 'n <= max_mem_access . (t : AccessType(ext
   }
 }
 
-/* dispatches to MMIO regions or physical memory regions depending on physical memory map */
-function checked_mem_read forall 'n, 0 < 'n <= max_mem_access . (t : AccessType(ext_access_type), paddr : xlenbits, width : atom('n), aq : bool, rl : bool, res: bool, meta : bool) -> MemoryOpResult((bits(8 * 'n), mem_meta)) =
-  if   within_mmio_readable(paddr, width)
-  then MemoryOpResult_add_meta(mmio_read(t, paddr, width), default_meta)
-  else if within_phys_mem(paddr, width)
-  then match ext_check_phys_mem_read(t, paddr, width, aq, rl, res, meta) {
-    Ext_PhysAddr_OK()     => phys_mem_read(t, paddr, width, aq, rl, res, meta),
-    Ext_PhysAddr_Error(e) => MemException(e)
-  } else match t {
-    Execute()  => MemException(E_Fetch_Access_Fault()),
-    Read(Data) => MemException(E_Load_Access_Fault()),
-    _          => MemException(E_SAMO_Access_Fault())
-  }
+// Check if access is permitted according to PMPs and PMAs.
```

Comment:
```suggestion
/* Check if access is permitted according to PMPs and PMAs. */
```
Other functions seem to use this comment style - I forget what is preferred.

Reply:
I was presuming that they're from a distant past when Sail didn't support `//`? I can't see any advantage to using `/*` when `//` is supported - it's more tedious to edit, looks worse IMO and means you can't use `/* */` to comment out blocks of code.

I'm hoping one day Sail's built in formatter will normalise all this stuff!

Reply:
I think it was just an arbitrary stylistic choice, as I think we have supported both comment styles for as long as sail-riscv has existed.

The block style comments are actually quite a bit easier to auto-format than line comments funnily enough.

Reply:
Marking resuled.